src/wasm.h

```diff
@@ -91,6 +91,11 @@ struct Name : public cashew::IString {
 // An index in a wasm module
 typedef uint32_t Index;
 
+inline Index toIndex(size_t index, size_t max) {
```

Comment:
How about if I create an overload with one parameter that includes your suggested assert?
 assert(index <= std::numeric_limits<uint32_t>::max())
Or should I just change two to one parameter? 

    On Thursday, May 19, 2016 8:04 PM, Alon Zakai <notifications@github.com> wrote:
 

 In src/wasm.h:> @@ -91,6 +91,11 @@ struct Name : public cashew::IString {
>  // An index in a wasm module
>  typedef uint32_t Index;
>  
> +inline Index toIndex(size_t index, size_t max) {
I'm not sure about the second parameter. It does help in some cases, but we may not always have that information around, and we'd need to write a lot of getFunction()->getNumLocals(). I suggest just asserting on just the size of the type, assert(index <= std::numeric_limits<uint32_t>::max()).—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub